firebase-functions/functions/src/firebase/calculateResult.ts

```diff
@@ -34,3 +43,8 @@ async function getCurrentRoomId(playerId: string): Promise<string> {
   const currentPlayer = await admin.firestore().collection(collections.players).doc(playerId).get();
   return currentPlayer.data()?.room;
 }
+
+async function getNumberOfPlayers(roomId: string): Promise<number> {
+  return (await admin.firestore().collection(collections.players).where('room', '==', roomId).get())
```

Comment:
The same database call + a filter is done in line 38. If you want I suggest to only call once and get all and filter out `isObserver` (on line 39 now) without another call then.
Leave it if you want, I don't think there will be a measurable performance impact, I just find it more elegant.

Reply:
you right!